project/src/main/java/com/google/sps/servlets/UserProfileUpdateServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.User;
+import com.google.sps.utilities.Constants;
+import com.google.sps.utilities.DatabaseWrapper;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.Set;
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonArrayBuilder;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles user profile edits. */
+@WebServlet("/profile-update")
+public class UserProfileUpdateServlet extends HttpServlet {
+  private static final String email = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+  private DatabaseWrapper databaseWrapper = new DatabaseWrapper(Constants.DB_INSTANCEID, Constants.DB_DATABASEID);
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    User user = databaseWrapper.readUserFromEmail(email);
+    String userJson = Json.createObjectBuilder()
+        .add("name", user.getName())
+        .add("email", email)
+        .add("interests", createJsonArray(user.getInterests()))
+        .add("skills", createJsonArray(user.getSkills()))
+        .build()
+        .toString();
+    response.setContentType("application/json;");
+    response.setCharacterEncoding("UTF-8");
+    response.getWriter().println(userJson);
+  }
+
+  public JsonArray createJsonArray(Set<String> elements) {
```

Comment:
Does this need to be public? Maybe use private static?

Reply:
Done, moved it to the CommonUtils class because I expect we will need it several times as a helper anyways.